web/cobrands/fixmystreet-uk-councils/alloy.js

```diff
@@ -0,0 +1,147 @@
+        var all_tiles = this.getTileRange_(options.scope.bounds, options.scope.layer.maxExtent, options.scope.layer.map);
+        var rresp;
+        var start = new Date();
+        var max = all_tiles.length;
+        $(fixmystreet).trigger('alloy:start_request', [start, max]);
+        for (var i = 0; i < max; i++) {
+            var resp = new OpenLayers.Protocol.Response({requestType: "read"});
+            resp.start = start;
+            var url = this.getURL(all_tiles[i], options);
+            resp.priv = OpenLayers.Request.GET({
+                url: url, //options.url,
+                callback: this.createCallback(this.handleRead, resp, options),
+                params: options.params,
+                headers: options.headers
+            });
+            rresp = resp;
+        }
+        return rresp;
+    },
+
+    getURL: function(coords, options) {
+        return OpenLayers.String.format(options.base, {'layerid': options.layerid, 'layerVersion': options.layerVersion, 'z': 15, 'x': coords[0], 'y': coords[1]});
+    },
+
+    getTileRange_: function(bounds, maxExtent, map) {
+        var min = this.getTileCoord_([bounds.left, bounds.top], maxExtent, map, true);
+        var max = this.getTileCoord_([bounds.right, bounds.bottom], maxExtent, map, false);
+        var coords = [];
+        for (var i = min[0], ii = max[0]; i <= ii; ++i) {
+          for (var j = min[1], jj = max[1]; j <= jj; ++j) {
+              coords.push([i,j]);
+          }
+        }
+        return coords;
+    },
+
+    getTileCoord_: function(bounds, maxExtent, map, reverse) {
+        var origin = new OpenLayers.LonLat(maxExtent.left, maxExtent.top);
+        var resolution = map.getResolutionForZoom(3);
+
+        var adjustX = reverse ? 0.5 : 0;
+        var adjustY = reverse ? 0 : 0.5;
+        var xFromOrigin = Math.floor((bounds[0] - origin.lon) / resolution + adjustX);
+        var yFromOrigin = Math.floor((bounds[1] - origin.lat) / resolution + adjustY);
+        var tileCoordX = Math.floor(xFromOrigin / 512);
+        var tileCoordY = Math.floor(yFromOrigin / 512) * -1;
+
+        if (reverse) {
+            tileCoordX -= 1;
+            tileCoordY -= 1;
+        }
+
+        return [ tileCoordX, tileCoordY ];
+    }
+});
+
+OpenLayers.Strategy.Alloy = OpenLayers.Class(OpenLayers.Strategy.FixMyStreet, {
+    count: 0,
+    max: 0,
+    requestStart: 0,
+    initialize: function(name, options) {
+        OpenLayers.Strategy.FixMyStreet.prototype.initialize.apply(this, arguments);
+        $(fixmystreet).on('alloy:start_request', this.newRequest.bind(this));
+    },
+    newRequest: function(evt, start, max) {
+      this.max = max;
+      this.requestStart = start;
+      this.count = 0;
+      this.layer.destroyFeatures();
+    },
+    merge: function(resp) {
+        // because we are issueing async requests it's possible that if someone moves the
+        // map we've triggered a new set of requests, in which case ignore the old ones.
```

Comment:
If you add an abort function to the Protocol, that then called resp.priv.abort() on all the requests (stored somewhere), then I guess this would not be needed? Probably doesn't matter

Reply:
I might leave that as a improvement for later rather than try and do it now.